exp/api/v1beta1/gcpmanagedmachinepool_types.go

```diff
@@ -37,6 +37,35 @@ type GCPManagedMachinePoolSpec struct {
 	// Scaling specifies scaling for the node pool
 	// +optional
 	Scaling *NodePoolAutoScaling `json:"scaling,omitempty"`
+	// NodeLocations is the list of zones in which the NodePool's
+	// nodes should be located.
+	// +optional
+	NodeLocations []string `json:"nodeLocations,omitempty"`
+	// ImageType is image type to use for this nodepool.
+	// +optional
+	ImageType *string `json:"imageType,omitempty"`
+	// InstanceType is name of Compute Engine machine type.
+	// +optional
+	InstanceType *string `json:"instanceType,omitempty"`
+	// DiskType is type of the disk attached to each node.
+	// +kubebuilder:validation:Enum=pd-standard;pd-ssd;pd-balanced
+	// +optional
+	DiskType *string `json:"diskType,omitempty"`
```

Comment:
When we have a enum like this we normally create a new type (from a string in this case) and declare constants for each of the possible values. Like this: 

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/2e5725a57078e0778939e7105c4303b5607a7377/exp/api/v1beta1/gcpmanagedcontrolplane_types.go#L110-L120